daemon/api_test.go

```diff
@@ -104,7 +104,22 @@ func (s *apiSuite) SetUpTest(c *check.C) {
 	s.overlord = &fakeOverlord{
 		configs: map[string]string{},
 	}
-	s.daemon = newTestDaemon(c)
+	s.d = nil
+}
+
+func (s *apiSuite) TearDownTest(c *check.C) {
+	s.d = nil
+}
+
+func (s *apiSuite) daemon(c *check.C) *Daemon {
+	if s.d != nil {
```

Comment:
Because that was suggested earlier. It also allows us to expand this to build a more complete daemon without changing the requirement that ``s.daemon(c)`` can be called only once. I have no strong feelings for this so if you want we can remove tihs.